lib/modules/ens/index.js

```diff
@@ -215,101 +191,100 @@ class ENS {
-          "Resolver": {
-            "deploy": false
-          },
-          "FIFSRegistrar": {
-            "deploy": false
-          }
+        "ENSRegistry": {
+          "address": "0xe7410170f87102DF0055eB195163A03B7F2Bff4A",
+          "silent": true
+        },
+        "Resolver": {
+          "deploy": false
+        },
+        "FIFSRegistrar": {
+          "deploy": false
         }
       },
       "livenet": {
-        "contracts": {
-          "ENSRegistry": {
-            "address": "0x314159265dd8dbb310642f98f50c066173c1259b",
-            "silent": true
-          },
-          "Resolver": {
-            "deploy": false
-          },
-          "FIFSRegistrar": {
-            "deploy": false
-          }
+        "ENSRegistry": {
+          "address": "0x314159265dd8dbb310642f98f50c066173c1259b",
+          "silent": true
+        },
+        "Resolver": {
+          "deploy": false
+        },
+        "FIFSRegistrar": {
+          "deploy": false
         }
       }
     };
-    config.testnet = config.ropsten;
+    knownConfigs.testnet = knownConfigs.ropsten; // Synonyms
+    knownConfigs.mainnet = knownConfigs.livenet; // Synonyms
+
+    if (knownConfigs[this.env]) {
+      self.ensConfig = utils.recursiveMerge(self.ensConfig, knownConfigs[this.env]);
+    }
 
-    if (this.registration && this.registration.rootDomain) {
-      // Register root domain if it is defined
-      const rootNode = namehash.hash(this.registration.rootDomain);
-      config.default.contracts['FIFSRegistrar'] = {
-        "deploy": true,
-        "silent": true,
-        "args": ["$ENSRegistry", rootNode],
-        "onDeploy": [
+    async.waterfall([
+      function registry(next) {
+        self.events.request('deploy:contract', self.ensConfig.ENSRegistry, (err, _receipt) => {
+          return next(err);
+        });
+      },
+      function resolver(next) {
+        self.ensConfig.Resolver.args = [self.ensConfig.ENSRegistry.deployedAddress];
+        self.events.request('deploy:contract', self.ensConfig.Resolver, (err, _receipt) => {
+          return next(err);
+        });
+      },
+      function registrar(next) {
+        if (!self.registration || !self.registration.rootDomain) {
+          return next();
+        }
+        const registryAddress = self.ensConfig.ENSRegistry.deployedAddress;
+        const resolverAddress = self.ensConfig.Resolver.deployedAddress;
+        const rootNode = namehash.hash(self.registration.rootDomain);
+        const contract = self.ensConfig.FIFSRegistrar;
+        contract.args = [registryAddress, rootNode];
+        contract.onDeploy = [
           `ENSRegistry.methods.setOwner('${rootNode}', web3.eth.defaultAccount).send({from: web3.eth.defaultAccount}).then(() => {
-            ENSRegistry.methods.setResolver('${rootNode}', "$Resolver").send({from: web3.eth.defaultAccount});
+            ENSRegistry.methods.setResolver('${rootNode}', "${resolverAddress}").send({from: web3.eth.defaultAccount});
```

Comment:
if we are still using the contracts config for this, then I think can be `ENSRegistry.methods.setResolver('${rootNode}', "$Resolver").send({from: web3.eth.defaultAccount});`

Reply:
Oh, yeah, you are right, it would be cleaner and we would have more control over it.

Reply:
I tried and ENSRegistry is not in the same global context, so it's harder than expected to do as you propose. I will do that in another PR to not burden this PR too much